### ISSUE TITLE
The selected option in the "Sessions" dropdown behaves poorly

### ISSUE THREAD

**Issue by J8cob1** [capstoneteamb/capstone360#104]
I noticed that the selected session the Sessions dropwdown doesn't always match the session currently being displayed in the dashboard.

- If I create a new session (using the Add Session button), the dashboard seems to load the new session, but the selected session in the "Sessions" dropdown doesn't get updated. 
  - **Example:** If I am in the professor dashboard viewing Rick Darren's Spring 2019 session (such that the selected option in the Sessions dropdown is "Spring 2019 (Rick Darren)") and I create a new session (lets say, a Fall 2022 session for George Portland), the dashboard appears to load the newly created session (I presume it does at least, because I don't see any teams being displayed), but the selected session in the Sessions dropdown remains "Spring 2019 (Rick Darren)". The selected item should have been updated to "Fall 2022 (George Portland)".
  - **Solution Idea:** update the sessions dropdown whenever we try to load a new session in the professor dashboard in a way that doesn't involve clicking a session in the "Selected Session" dropdown
- If I log into the professor dashboard, I am taken to one of my sessions, but the selected item in the Sessions dropdown isn't updated.
  - **Example:** if I log in to the professor dashboard as MCECS user "kaymark" (after using the dashboard, navigating to localhost:8000 and clicking the "Professor Login" button to get back to the professor dashboard) and I have a Spring 2022 session, the Spring 2022 session appears to load in the dashboard, but the selected session in the "Sessions" dropdown will be something else ("Spring 2019 (George Portland)" for example--perhaps whatever was selected the last time we loaded the dashboard)
  - **Solution Idea:** update the sessions dropdown whenever we try to load a new session in the professor dashboard in a way that doesn't involve clicking a session in the "Selected Session" dropdown (or at least, on login)
- If you remove a session, the dashboard seems to load the session that is listed immediately before it in the sessions dropdown, but the session does not appear to be selected in the Sessions dropdown itself (it becomes blank)
  - **Example:** [to get the idea across] If you have four sessions in you db: Spring 2019, Summer 2019, Winter 2020 and Spring 2020, and you remove the last session (Spring 2020), the Winter 2020 session appears to be loaded in the dashboard, but no option will be shown in the selected sessions dropdown (it will instead be blank)
  - **Solution Idea:** Update the selected item in the sessions dropdown when deleting an item (or whenever a new session is loaded in some way that doesn't involve clicking a session in the sessions dropdown)
- [Cosmetic thing that would be nice] Order the sessions in the sessions dropdown alphabetically (it might look kind of confusing if you create a Spring 1999 session [for whatever reason] and see that your 1999 session is comes after your Spring 2020 session)

**Comment by J8cob1** [comment]
Actually, I might try to verify this a bit more before I file an issue.

**Comment by J8cob1** [comment]
I see an issue on the master branch. Example of something that can go wrong:

When I log in, I am redirected to one of my sessions, Session 10 - "Spring 2019 (Jacob Crisan)") for example, but the currently selected item in the selected option dropdown is "Winter 2019 (George Portland)"

**Comment by J8cob1** [comment]
We may just need to throw in some code to make sure it gets updated

**Comment by J8cob1** [comment]
I was checking Long's styling branch and it looks like the sessions dropdown also misbehaves right after you remove a session (the last session in the app--that is, the one that was most recently created--is loaded in the dashboard, but the dropdown doesn't indicate which one is selected. This might also be in the main branch

**Comment by J8cob1** [comment]
There also might be a problem after you add a new session

**Comment by pdx-cs-lbryan** [comment]
It would be helpful if you could specify the issue at the top. This structure is a little hard to follow.

* What is the issue/issues on master? Examples of expected and observed behavior would be helpful
* (How) can you reproduce the issue?
* Do you have any suggestions for a fix?

**Comment by J8cob1** [comment]
Done, but we may not have enough time

**Comment by J8cob1** [comment]
We might also consider ordering the session (so that they come in alphabetical order)

**Comment by J8cob1** [comment]
Is this fixed now?

**Comment by pdx-cs-lbryan** [comment]
edit: sorry, master had old code. ignore the below The first and second seem fine. The third you'd need to weigh in on?

----

seem to still be having some issues what what you described here, though number 2 seems fine. you might try going into the browser debugger and seeing what's going on with the javascript an wrote.

Actually selecting a course with the selection option seems fine

**Comment by J8cob1** [comment]
The third thing doesn't seem fixed, though it behaves differently. What it seems to be happening now, after I remove a session, is that the professor dashboard opens up the session that appeared right before it in the sessions dropdown, but the selected session in the sessions dropdown doesn't show that session as the currently selected session (it seems to show the last session in the list as the currently selected session instead).
